### ISSUE TITLE
Expected string default value for '--jbuilder'; got true (boolean)

### ISSUE THREAD

**Issue by MarskalGroup** [rails/rails#27229]
### Steps to reproduce

rails g   #Pretty much any rails g commands show this error

### Expected behavior
I expect no error

### Actual behavior
the following message appears ==> Expected string default value for '--jbuilder'; got true (boolean)

Otherwise everything seems to function properly

### System configuration
**Rails version**: 4.2.7.1 (and others)

**Ruby version**: 2.2.5 (and others)

**Comment by sgrif** [comment]
Looks like this was a change in Thor. https://github.com/erikhuda/thor/blob/master/CHANGELOG.md#0192-release-2016-11-26

It's a warning only, not an error -- And it does not occur in Rails 5, so I do not think we need to worry about this issue.

**Comment by slbug** [comment]
happens in rails 5.0.0.1

<pre>
$ rails g migration anything
Expected string default value for '--test-framework'; got false (boolean)
Expected string default value for '--jbuilder'; got true (boolean)
</pre>

bundle list

<pre>
Gems included by the bundle:
  * actioncable (5.0.0.1)
  * actionmailer (5.0.0.1)
  * actionpack (5.0.0.1)
  * actionview (5.0.0.1)
  * activejob (5.0.0.1)
  * activemodel (5.0.0.1)
  * activerecord (5.0.0.1)
  * activesupport (5.0.0.1)
  * acts_as_list (0.8.2)
  * addressable (2.5.0)
  * airbrussh (1.1.1)
  * apartment (1.2.0 80de24f)
  * arel (7.1.4)
  * bcrypt (3.1.11)
  * bootscale (0.6.1 8f5193a)
  * brakeman (3.4.1)
  * builder (3.2.2)
  * bullet (5.4.3 2c96b20)
  * bundler (1.12.5)
  * byebug (9.0.6)
  * capistrano (3.7.1)
  * capistrano-bower (1.1.0)
  * capistrano-bundler (1.2.0)
  * capistrano-dotenv-tasks (0.1.4)
  * capistrano-gulp (0.0.2)
  * capistrano-harrow (0.5.3)
  * capistrano-nginx (1.0.0)
  * capistrano-npm (1.0.2)
  * capistrano-postgresql (4.2.1)
  * capistrano-rails (1.2.0)
  * capistrano-rvm (0.1.2)
  * carrierwave (0.11.2)
  * carrierwave-imageoptimizer (1.4.0)
  * concurrent-ruby (1.0.3)
  * connection_pool (2.2.1)
  * consul (0.12.3)
  * devise (4.2.0)
  * devise_invitable (1.7.0 90318ed)
  * dotenv (2.1.1)
  * dotenv-rails (2.1.1)
  * edge_rider (0.3.1)
  * erubis (2.7.0)
  * faraday (0.9.2)
  * ffi (1.9.14)
  * foreman (0.82.0)
  * globalid (0.3.7)
  * i18n (0.7.0)
  * image_optimizer (1.7.0)
  * jbuilder (2.6.1)
  * json (2.0.2)
  * jwt (1.5.6)
  * kaminari (0.17.0)
  * koala (2.4.0)
  * listen (3.1.5)
  * loofah (2.0.3)
  * mail (2.6.4)
  * memoizer (1.0.2)
  * method_source (0.8.2)
  * mime-types (3.1)
  * mime-types-data (3.2016.0521)
  * mimemagic (0.3.2)
  * mini_portile2 (2.1.0)
  * minitest (5.10.1)
  * monetize (1.4.0)
  * money (6.7.1)
  * money-rails (1.7.0)
  * msgpack (1.0.2)
  * multi_json (1.12.1)
  * multi_xml (0.6.0)
  * multipart-post (2.0.0)
  * net-scp (1.2.1)
  * net-ssh (3.2.0)
  * nio4r (1.2.1)
  * nokogiri (1.6.8.1)
  * oauth2 (1.2.0)
  * oj (2.18.0)
  * oj_mimic_json (1.0.1)
  * orm_adapter (0.5.0)
  * pg (0.19.0)
  * public_suffix (2.0.4)
  * puma (3.6.2)
  * rack (2.0.1)
  * rack-protection (1.5.3)
  * rack-test (0.6.3)
  * rails (5.0.0.1)
  * rails-dom-testing (2.0.1)
  * rails-html-sanitizer (1.0.3)
  * rails_12factor (0.0.3)
  * rails_serve_static_assets (0.0.5)
  * rails_stdout_logging (0.0.5)
  * railties (5.0.0.1)
  * rake (12.0.0)
  * rb-fsevent (0.9.8)
  * rb-inotify (0.9.7)
  * recaptcha (4.0.0)
  * redis (3.3.2)
  * redis-namespace (1.5.2)
  * responders (2.3.0)
  * rmagick (2.16.0)
  * rolify (5.1.0)
  * rollbar (2.13.3)
  * ruby_dep (1.5.0)
  * rufus-scheduler (3.3.1)
  * russian (0.6.0)
  * sidekiq (4.2.7)
  * sidekiq-cron (0.4.5)
  * sixarm_ruby_unaccent (1.1.1)
  * spring (2.0.0)
  * spring-watcher-listen (2.0.1)
  * sprockets (3.7.0)
  * sprockets-rails (3.2.0)
  * sshkit (1.11.5)
  * summernote-rails (0.8.2.0)
  * thor (0.19.4)
  * thread_safe (0.3.5)
  * tzinfo (1.2.2)
  * uniform_notifier (1.10.0)
  * warden (1.2.6)
  * websocket-driver (0.6.4)
  * websocket-extensions (0.1.2)
</pre>

**Comment by hirokishirai** [comment]
@slbug If you'll downgrade thor  to 0.19.1, this issue will fix on Rails5.

**Comment by realvjy** [comment]
![image](https://cloud.githubusercontent.com/assets/7387461/22977720/d1cb45e0-f3b5-11e6-8e26-6173e1bbbd2a.png)
 I'm using rails 5. Still get this warning @sgrif

**Comment by sevenseacat** [comment]
@realvjy did you downgrade thor to 0.19.1?

**Comment by realvjy** [comment]
thor 0.19.4 was locked in gemfile.lock. I'm checking. 
![image](https://cloud.githubusercontent.com/assets/7387461/22980360/33776b30-f3bf-11e6-95be-6b05c04e0951.png)
It works. Thank you @sevenseacat

**Comment by smelike** [comment]
> rails g --help                                                                                                                                        
> Running via Spring preloader in process 10245                                                                                                                                   
> Expected string default value for '--jbuilder'; got true (boolean)

**Comment by smelike** [comment]
Downgrade thor?
`gem install thor -v 0.19.1`

**Comment by smelike** [comment]
> > ➜  toy_app git:(master) ✗ rails generate scaffold User name:string email:string                                                                                                 
> Running via Spring preloader in process 13060                                                                                                                                   
> Expected string default value for '--jbuilder'; got true (boolean)                                                                                                              
>       invoke  active_record                                                                                                                                                     
>    identical    db/migrate/20170412074127_create_users.rb                                                                                                                       
>    identical    app/models/user.rb                                                                                                                                              
>       invoke    test_unit                                                                                                                                                       
>    identical      test/models/user_test.rb                                                                                                                                      
>    identical      test/fixtures/users.yml                                                                                                                                       
>       invoke  resource_route                                                                                                                                                    
>        route    resources :users                                                                                                                                                
>       invoke  scaffold_controller                                                                                                                                               
>     conflict    app/controllers/users_controller.rb                                                                                                                             
>   Overwrite /home/coding/workspace/toy_app/app/controllers/users_controller.rb? (enter "h" for help) [Ynaqdh] y

**Comment by antonpaisov** [comment]
happens to me on rails 5.0.3 too, thor downgrade indeed helps, but that doesn't sound like a legit fix :)

**Comment by utilum** [comment]
Would it not be a legit fix in master Gemfile pending ... ?

**Comment by jaredbeck** [comment]
> .. it does not occur in Rails 5 ..

Happens to me in 5.0.4. Reproduce with `bin/rails g`.

@rafaelfranca [says this is something rails has to fix][1], so I think this issue can be re-opened? @sgrif 

[1]: https://github.com/erikhuda/thor/issues/538#issuecomment-279270777

**Comment by rafaelfranca** [comment]
This is already fixed on latest jbuilder, just make sure you updated it

**Comment by ivanovaleksey** [comment]
Issue with jbuilder seems to be fixed but I still have
```
$ rails g
Expected string default value for '--decorator'; got true (boolean)
Expected string default value for '--decorator'; got true (boolean)
```

**Comment by kirkelifson** [comment]
Fixed by downgrading thor to 0.19.2 (required by jquery-rails and railties)
